installscripts/jazz-terraform-unix-noinstances/ecs.tf

```diff
@@ -0,0 +1,389 @@
+resource "aws_iam_role_policy" "ecs_execution_policy" {
+  count = "${var.dockerizedJenkins}"
+  name = "${var.envPrefix}_ecs_execution_policy"
+  role = "${aws_iam_role.ecs_execution_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role" "ecs_execution_role" {
+  count = "${var.dockerizedJenkins}"
+  name = "${var.envPrefix}_ecs_execution_role"
+
+ assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_cloudwatch_log_group" "ecs_fargates_cwlogs" {
+  count = "${var.dockerizedJenkins}"
+  name = "${var.envPrefix}_ecs_log"
+  retention_in_days = 7
+}
+
+resource "null_resource" "ecs_securitygroups" {
```

Comment:
The current ECS implementation is using the existing security group for setting up both Fargate and ALB. For jenkins (8080), Gitlab(80) and SonarQube(9000) must contain the security group ID as the inbound rule for those ports. 

The reason why aws cli is because, what if there is already security group attached to those ports, terraform will throw error if we use aws_security_group_rule. So aws cli has been used with on_failure="continue".

Reply:
Yeah, created fresh security group

Reply:
Awesome, thanks!